packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js

```diff
@@ -91,36 +88,67 @@ export let ScrollGradient = React.forwardRef(
       endHorizontalRef.current.style.backgroundImage = `linear-gradient(-90deg, ${color} 10%, transparent)`;
     }, [color, gradientRight, gradientBottom]);
 
-    const updateScrollState = throttle(() => {
-      const updatedVerticalVal = getScrollState(
-        scrollContainer.current,
-        ScrollDirection.Y
-      );
-      const updatedHorizontalVal = getScrollState(
-        scrollContainer.current,
-        ScrollDirection.X
+    const setGradientOnIntersection = (entry, gradientRef) => {
+      if (gradientRef.current) {
+        if (entry.isIntersecting) {
+          gradientRef.current.style.opacity = 0;
+          gradientRef.current.style.display = 'none';
+          gradientRef?.current.setAttribute('aria-hidden', false);
+        } else {
+          gradientRef.current.style.opacity = 1;
+          gradientRef.current.style.display = 'block';
+          gradientRef?.current.setAttribute('aria-hidden', true);
+        }
+      }
+    };
+
+    useEffect(() => {
+      const observer = new IntersectionObserver(
```

Comment:
I did some refactoring here to use `IntersectionObserver` rather than a scroll event for tracking when the gradients should become visible.